src/legacy/core_plugins/vega/public/index.ts

```diff
@@ -17,20 +17,7 @@
  * under the License.
  */
 
+import { VegaPlugin } from './plugin';
+import { plugins } from './setup';
 
-import 'ngreact';
-
-import { wrapInI18nContext } from 'ui/i18n';
-import { uiModules } from 'ui/modules';
-const module = uiModules.get('kibana/vega', ['react']);
-
-import { VegaHelpMenu } from './vega_help_menu';
-import { VegaActionsMenu } from './vega_action_menu';
-
-module.directive('vegaActionsMenu', function (reactDirective) {
-  return reactDirective(wrapInI18nContext(VegaActionsMenu));
-});
-
-module.directive('vegaHelpMenu', function (reactDirective) {
-  return reactDirective(wrapInI18nContext(VegaHelpMenu));
-});
+export const plugin = new VegaPlugin().setup(plugins);
```

Comment:
In Embeddables I was doing something like this:

```ts
import { plugin } from '.';
import { npSetup, npStart } from 'ui/new_platform';

const pluginInstance = plugin({} as any);
export const setup = pluginInstance.setup(npSetup.core);
export const start = pluginInstance.start(npStart.core);
```

Maybe you could do something like:

```ts
import { VegaPlugin } from './plugin';
import { npSetup } from 'ui/new_platform';
import { plugins } from './setup';

const plugin = new VegaPlugin();
export const setup = plugin.setup(npSetup.core, plugins);
```

And rename this file to `legacy.ts`, because we will remove this file, when actually moving this to `/src/plugins` and in `index.ts` there should be a plugin initializer.

---

I don't know if you use any shim as an example, but maybe take a look at embeddables: https://github.com/streamich/kibana/tree/shim-embeddable/src/legacy/core_plugins/embeddable_api/public/shim/public

Specifically, `plugin.ts`, `index.ts` and `legacy.ts` files you can make like there.